code-challenges.test.js

```diff
@@ -54,6 +127,52 @@ var vowelTester2 = "academy"
 var vowelTester3 = "challenges"
 // Expected output: 2
 
-
+describe ("vowelTester", () => {
+    it ("logs the index of the first vowel in a string", () => {
+        //expect the argument to result in a number
+        expect(vowelTester(vowelTester1)).toEqual(1)
+        expect(vowelTester(vowelTester2)).toEqual(0)
+        expect(vowelTester(vowelTester3)).toEqual(2)
+    })
+})
+//ReD
 
 // b) Create the function that makes the test pass.
+
+// create a function that determines if a letter is a vowel using a switch statement
+//if the input string is a vowel, return true, otherwise return false
+const isVowel = (string) => {
+    switch (string) {
+        case "a":
+            return true
+            break;
+        case "e":
+            return true
+            break;
+        case "i":
+            return true
+            break;
+        case "o":
+            return true
+            break;
+        case "u":
+            return true
+            break;
+        default:
+            false
+            break;
+    }
+}
+
+//Declare function vowelTester that takes in a string 
+const vowelTester = (string) => {
+    //use a for loop to iterate through the string and return the index of the first vowel
+    for (let i = 0; i < string.length; i++) {
+        //Use an if statement combined with isVowel to determine if the letter at the current index is a Vowel, and if it is log it/return it
+        if ((isVowel(string[i]))) {
+            console.log(i)
+            return i
+        }
+        
+    }
+}
```

Comment:
Interesting approach. I think some logical operators might shorten your code significantly.

Reply:
I agree, I came across the following recently and i think it cleans it up nicely.

const vowelTester = (string) => {
  return string.search(/[aeiou]/gi)
}

Reply:
Yeah that is a library called Regular Expressions. It is a set of string methods. That works. There are also shorter programmatic solutions.

Reply:
You've caught my attention, what are the shorter solutions?